### PR TITLE
ldc2.conf: %%ldcconfigpath%% support added

### PR DESCRIPTION
**Pull request by denizzzka**
Fix https://github.com/ldc-developers/ldc/issues/4714

### CODE REVIEW

**Review by thewilsonator — COMMENTED** [review]
Thanks, could you add this to the changelog as well as some documentation.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #### Big news
 - Android: NDK for prebuilt package bumped from r26d to r27. (#4711)
+- ldc2.conf: %%ldcconfigpath%% placeholder added - specifies the directory where current configuration file is located.
```
Comment (thewilsonator) [inline CHANGELOG.md:5]: please also add the PR number to the end (see the above entry for android)

**Comment by thewilsonator** [comment]
> There is no documentation for %%ldcbinarypath%% placeholder, so I don't know where and what to write

Hmm, will wait for others to hopefully give some guidance.

File: CHANGELOG.md (hunk 1 above)
Comment (denizzzka) [reply CHANGELOG.md]: Done

File: driver/configfile.d [hunk 2]
```diff
@@ -117,8 +150,7 @@ private:
     {
         switches.setDim(0);
         postSwitches.setDim(0);
-
-        immutable dBinDir = prepareBinDir(binDir);
+        const CfgPaths cfgPaths = { cfPath: cfPath, binDir: binDir };
```
Comment (kinke) [inline driver/configfile.d:153]: Hey Dennis, sorry for the delay. - Please convert these paths once here, not for every `replacePlaceholders()` call.

That `prepareBinDir()` is an awful name - apparently its job is to normalize the slashes in the path to forward slashes. No idea why or whether that's necessary, but oh well. If you wanna keep that for the new variable, feel free to, but then we need to rename the function.

Your `baseDir()` addition shouldn't be required; there's a static overloaded `dmd.root.filename.FileName.path()` function that does this already.

Comment (denizzzka) [reply driver/configfile.d]: Done

**Review by kinke — APPROVED** [review]
Thanks!

**Pull request merged**